src/core/stdc/stdlib.d

```diff
@@ -157,11 +157,11 @@ else
 // functions using these already have to be @trusted, allowing them to
 // call @system stuff anyway.
 ///
-void*   malloc(size_t size) pure;
+void*   malloc(size_t size);
```

Comment:
OK. Next step, please define a function `pureMalloc` that sets `errno` back to where it was for the current thread.